runtime/master/src/main/java/org/apache/nemo/runtime/master/resource/ResourceSpecification.java

```diff
@@ -29,22 +31,25 @@
   private final String containerType;
   private final int capacity;
   private final int memory;
-  private final int poisonSec; // -1 if this resources is not poisoned
+  private final Optional<Double> maxOffheapRatio;
+  private final Optional<Integer> poisonSec; // -1 if this resources is not poisoned
```

Comment:
Let's change this to `OptionalDouble` and `OptionalInt` which provides more features for the specific data types

Reply:
done! thanks.